test/integration/dnsrecord/dnsrecord_test.go

```diff
@@ -0,0 +1,488 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dnsrecord_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/route53"
+	awsinstall "github.com/gardener/gardener-extension-provider-aws/pkg/apis/aws/install"
+	"github.com/gardener/gardener-extension-provider-aws/pkg/aws"
+	awsclient "github.com/gardener/gardener-extension-provider-aws/pkg/aws/client"
+	dnsrecordctrl "github.com/gardener/gardener-extension-provider-aws/pkg/controller/dnsrecord"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/extensions"
+	gardenerutils "github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/test/framework"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var (
+	accessKeyID     = flag.String("access-key-id", "", "AWS access key id")
+	secretAccessKey = flag.String("secret-access-key", "", "AWS secret access key")
+)
+
+func validateFlags() {
+	if len(*accessKeyID) == 0 {
+		panic("need an AWS access key id")
+	}
+	if len(*secretAccessKey) == 0 {
+		panic("need an AWS secret access key")
+	}
+}
+
+var _ = Describe("DNSRecord tests", func() {
+	var (
+		ctx = context.Background()
+
+		logger    *logrus.Entry
+		awsClient *awsclient.Client
+		testEnv   *envtest.Environment
+		mgrCancel context.CancelFunc
+		c         client.Client
+
+		testName = fmt.Sprintf("aws-dnsrecord-it--%s", randomString())
+		zoneName = testName + ".gardener.cloud"
```

Comment:
There is absolutely no issue with hardcoding it. You can create the zone "<id>.gardener.cloud" in any account, even duplicated names would be ok (although since the test ID is generated, no duplication is possible in this case). The records created are of course not resolvable, but our integration test doesn't really care. There is no real need to use an own domain / hosted zone, it simply doesn't matter.

Reply:
Ah got it. I wasn't aware, that you can create hosted zones with freely chosen names anywhere.
But now that you say it, it makes perfect sense 👍 
Then we can keep it of course.